File: python/atoi.py
#!/usr/bin/env python
def IP2Int(ip):
    o = list(map(int, ip.split('.')))
    res = (16777216 * o[0]) + (65536 * o[1]) + (256 * o[2]) + o[3]
    return res


def Int2IP(ipnum):
    o = []
    for i in range(0, 4):
        o.append(ipnum % 256)
        ipnum /= 256
    return ".".join(map(str, o[::-1]))

if __name__ == "__main__":
    import sys
    if "." in sys.argv[1]:
        print(IP2Int(sys.argv[1]))
    else:
        print(Int2IP(int(sys.argv[1])))

File: go/atoi.go
package main

import (
    "math"
    "os"
    "fmt"
    "strings"
    "strconv"
)

func IP2Int(ip string) int {
    var res int
    for i, octet := range strings.Split(ip, ".") {
        o, err := strconv.Atoi(octet)
        if err != nil {
            return 0
        }
        res += o * int(math.Pow(256, float64(3 - i)))
    }
    return res
}

func Int2IP(ipnum int) string {
    octets := make([]string, 4)
    for i := 3; i >= 0; i-- {
        octets[i] = strconv.Itoa(ipnum % 256)
        ipnum = ipnum / 256
    }
    return strings.Join(octets, ".")
}

func main(){
    if strings.Contains(os.Args[1], ".") {
        fmt.Println(IP2Int(os.Args[1]))
    } else {
        ipnum, err := strconv.Atoi(os.Args[1])
        if err != nil {
            fmt.Println("Bad input")
            return
        }
        fmt.Println(Int2IP(ipnum))
    }
}